test/OpenTelemetry.Instrumentation.SqlClient.Tests/SqlClientInstrumentationOptionsTests.cs

```diff
@@ -94,5 +94,27 @@ static SqlClientInstrumentationOptionsTests()
             Assert.Equal(expectedInstanceName, activity.GetTagValue(SemanticConventions.AttributeDbMsSqlInstanceName));
             Assert.Equal(expectedPort, activity.GetTagValue(SemanticConventions.AttributeNetPeerPort));
         }
+
+        [Theory]
+        [InlineData(true, "Data Source=127.0.0.1,5433; User ID=sa; Password=Pass@word", "sa")]
+        [InlineData(true, "Data Source=127.0.0.1,5433; UID=sa; Password=Pass@word", "sa")]
+        [InlineData(true, "Data Source=127.0.0.1,5433; user=sa; Password=Pass@word", "sa")]
+        [InlineData(true, "Data Source=127.0.0.1,5433; Password=Pass@word", null)]
+        [InlineData(false, "Data Source=127.0.0.1,5433; User ID=sa; Password=Pass@word", null)]
+        public void SqlClientInstrumentationOptions_EnableConnectionLevelAttributes_DBUser(
+            bool enableConnectionLevelAttributes,
+            string connectionString,
+            string userId)
+        {
+            var source = new ActivitySource("sql-client-instrumentation");
+            var activity = source.StartActivity("Test Sql Activity");
+            var options = new SqlClientInstrumentationOptions
+            {
+                EnableConnectionLevelAttributes = enableConnectionLevelAttributes,
+            };
+            options.AddDBUserToActivity(connectionString, activity);
+
+            Assert.Equal(userId, activity.GetTagValue(SemanticConventions.AttributeDbUser));
```

Comment:
I am a bit confused.. This assertion will only be true **If** enableConnectionLevelAttributes is true right?

Reply:
Updated

Reply:
doesn't seem correct. The test is passing always means userid is *always* set? It should only be set when user explicitly opts-in right?

Reply:
I put the opt-in checker from `AddDBUserToActivity` to [SqlClientDiagnosticListener](https://github.com/open-telemetry/opentelemetry-dotnet/blob/0940c0e12ce92555cfba887dbc0321615a67fcdf/src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs#L102), so basically `AddDBUserToActivity` will only try to extract user id from connection string.